### ISSUE TITLE
Feature request: Support Scheduling within the Salesforce Input Plugin

### ISSUE THREAD

**Issue by marc-sensenich** [logstash-plugins/logstash-input-salesforce#19]
As a feature, it'd be beneficial if the Salesforce Input plugin supported scheduling similar to other plugins such as the File or JDBC input plugins

**Comment by Mauro-Rizzi** [comment]
+1 I really need this to integrate this plugin with the rest of my pipelines

**Comment by marc-sensenich** [comment]
@Mauro-Rizzi I've reopened https://github.com/logstash-plugins/logstash-input-salesforce/pull/30 to add this feature to the plugin

**Comment by MatheusGelinskiPires** [comment]
Hello folks,
Do we have a target date to release this feature?

**Comment by fransf-wtax** [comment]
@MatheusGelinskiPires This plugin seems to have been abandoned. Last change on the main branch was over a year ago. I've started [a post](https://discuss.elastic.co/t/logstash-input-salesforce-still-alive-process-for-abandoned-plugin-projects/361928) on the Logstash forum to try to understand how ownership of plugins works and if it's possible for someone else to take over an abandoned plugin. While it's open source and you can bake your own and install it, installing "official" plugins is a lot easier so I do want to try to see if we can resurrect this plugin and get a new release out with some of the open PRs merged.

**Comment by jsvd** [comment]
(copying from discuss)

> hi @frans-wtax, thanks for taking interest and volunteering to maintain the salesforce input!
>
> The plugins under the logstash-plugins github org are under different levels of maintenance depending on demand, given that we're a small team and can't actively maintain all 200 of them.
>
> Logstash ships about 90 of those, which we keep a closer look on, but there are plugins that are currently being maintained by external contributors.
> If you wish to maintain this plugin we can make you an external contributor to the plugin on Github, and make you a gem owner on rubygems.org so you can publish new versions.
>
> If you're ok with that we can start working on giving you the necessary permissions. Elastic is still here to help if you get blocked or need further assistance.

**Comment by fransf-wtax** [comment]
Thanks @jsvd ! I got the invite but it wants me to log in to https://elastic.okta.com/ for which I don't have a username/password.  I don't see a sign up link either. Am I missing something?

Thanks!
Frans

**Comment by jsvd** [comment]
Let me figure that out, Elastic has some enforcements on the github orgs, but they shouldn't affect the external contributors.

**Comment by flexitrev** [comment]
@fransf-wtax I've filed a ticket to have you added as an External Collaborator. We do have an agreement you'll need to sign -
https://www.elastic.co/contributor-agreement

**Comment by fransf-wtax** [comment]
@flexitrev Signed. Note that it will be easier for me to make contributions using my personal GitHub user (@fransflippo), so I've put that name on the form.

**Comment by fransf-wtax** [comment]
@flexitrev Do you need anything else from me? Thanks!

**Comment by jsvd** [comment]
@fransf-wtax sorry that it's taking so long, we're still working through some internal hoops to add you as an external contributor, we'll let you know when it's done asap.

**Comment by MatheusGelinskiPires** [comment]
> @fransf-wtax sorry that it's taking so long, we're still working through some internal hoops to add you as an external contributor, we'll let you know when it's done asap.

hello @jsvd , could you please add me as well (if possible)? I already signed the contributor agreement.

**Comment by jsvd** [comment]
@fransflippo I sent you an invite to the repository instead of the whole org, can you check if this works?

**Comment by fransflippo** [comment]
Thanks, @jsvd that seems to have worked!

**Comment by jsvd** [comment]
@fransflippo awesome! what is the email you'd like to add to the rubygems gem ownership list?

**Comment by fransflippo** [comment]
@jsvd Can I send that to you via a private channel (e.g. Twitter DM)?
My RubyGems profile is fransflippo, in case that’s enough. 
Obrigado!

**Comment by jsvd** [comment]
For either the good or the bad, emails aren't really hidden on github. Is it the email on this commit? https://github.com/fransflippo/sfdx-ci-plugin/commit/7b24222530fdef3f888b0a076266b5912d56c469.patch

**Comment by fransf-wtax** [comment]
@jsvd No, that's yet another one! But now I know why I get so much spam on that address 😂 

I'll DM you, thanks.

**Comment by fransf-wtax** [comment]
@jsvd Oh boy, I can't DM you because you're not following me and now Elon wants to upsell me to Premium 😞 . Would you mind following me: @fransflippo on X/Twitter? Thanks.
